From 0c8722d006c4768af0728ef1ea68bc75b4471c39 Mon Sep 17 00:00:00 2001
From: ivanmeler <i_ivan@windowslive.com>
Date: Tue, 4 Oct 2022 10:22:57 +0000
Subject: [PATCH] Disable init failure for non bpf usecases

---
 netd/NetdUpdatable.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/netd/NetdUpdatable.cpp b/netd/NetdUpdatable.cpp
index f0997fc15..446cd4496 100644
--- a/netd/NetdUpdatable.cpp
+++ b/netd/NetdUpdatable.cpp
@@ -29,10 +29,6 @@ int libnetd_updatable_init(const char* cg2_path) {
 
     android::net::gNetdUpdatable = android::net::NetdUpdatable::getInstance();
     android::netdutils::Status ret = android::net::gNetdUpdatable->mBpfHandler.init(cg2_path);
-    if (!android::netdutils::isOk(ret)) {
-        LOG(ERROR) << __func__ << ": BPF handler init failed";
-        return -ret.code();
-    }
     return 0;
 }
 
-- 
2.34.1

