From 3bf4dbfb2cbc33ea37a4a0c8f54689585982f6d1 Mon Sep 17 00:00:00 2001
From: ivanmeler <i_ivan@windowslive.com>
Date: Tue, 4 Oct 2022 10:30:27 +0000
Subject: [PATCH] Remove runtime exception on legacy when removing packages

---
 core/java/android/app/LoadedApk.java | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/core/java/android/app/LoadedApk.java b/core/java/android/app/LoadedApk.java
index 59d518e6..7f6b4f1d 100644
--- a/core/java/android/app/LoadedApk.java
+++ b/core/java/android/app/LoadedApk.java
@@ -1797,9 +1797,6 @@ public final class LoadedApk {
                         if (mInstrumentation == null ||
                                 !mInstrumentation.onException(mReceiver, e)) {
                             Trace.traceEnd(Trace.TRACE_TAG_ACTIVITY_MANAGER);
-                            throw new RuntimeException(
-                                    "Error receiving broadcast " + intent
-                                            + " in " + mReceiver, e);
                         }
                     }
 
-- 
2.34.1

